marine.c

```diff
@@ -281,46 +293,38 @@ marine_write_specified_fields(packet_filter *filter, epan_dissect_t *edt, char *
 
     //char *output = (char *) g_malloc0(4096); // todo this can overflow
     int counter = 0;
-    gboolean has_value = FALSE;
     for (i = 0; i < fields->fields->len; ++i) {
         gchar *field = (gchar *) g_ptr_array_index(fields->fields, i);
         unsigned int fixed_index = GPOINTER_TO_UINT(g_hash_table_lookup(fields->field_indicies, field)) - 1;
 
         if (filter->macro_ids != NULL && (g_hash_table_contains(used_macros, filter->macro_ids + i) || (g_ptr_array_len(fields->field_values[fixed_index]) == 0 && !filter->last_in_macro[i]))) {
             continue;
         }
-        if (has_value) {
-            output[counter++] = fields->separator;
-        }
 
         if (NULL != fields->field_values[fixed_index]) {
             GPtrArray *fv_p;
             gchar *str;
             gsize j;
             fv_p = fields->field_values[fixed_index];
-            if (fields->quote != '\0') {
-                output[counter++] = fields->quote;
-            }
+            output[counter] = (gchar *) g_malloc0(get_field_length(fv_p));
+            int field_counter = 0;
 
             /* Output the array of (partial) field values */
             for (j = 0; j < g_ptr_array_len(fv_p); j++) {
                 str = (gchar *) g_ptr_array_index(fv_p, j);
                 for (char *p = str; *p != '\0'; p++) {
-                    output[counter++] = *p;
+                    output[counter][field_counter++] = *p;
```

Comment:
Why not `strncpy`?

Reply:
`strncpy` adds padding in case a NULL character is reached before the given string.